DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/use_library/part2_lib_func_call/src/use_library.cpp

```diff
@@ -98,12 +97,10 @@ int main() {
     std::terminate();
   }
 
-  // Compute the expected "golden" result
-  unsigned long gold = (unsigned long) kA * kB;
-  
   // Check the results
-  if (result_rtl != gold) {
-    std::cout << "FAILED: result (" << result_rtl << ") is incorrect! Expected " << gold << "\n";
+  unsigned long expected_result = (unsigned long) kA * kB;
```

Comment:
```suggestion
  uint64_t expected_result = (uint64_t ) kA * kB;
```
please remove unsigned long in the rest of this code and replace with uint64_t